src/plugins/cache.js

```diff
@@ -107,6 +127,21 @@ export default class Cache {
 
   async fetchManagers (...managers: Manager[]): Promise<Plugin[]> {
     let plugins = []
+    if (this.cache.node_version !== process.version) {
+      let lock = new Lock(this.out)
+      let handled = true
+
+      let downgrade = await lock.upgrade()
+      for (let manager of managers) {
+        handled = await manager.handleNodeVersionChange() && handled
+      }
+      await downgrade()
+
+      if (handled) {
+        this.cache.node_version = process.version
+        this.constructor.updated = true
```

Comment:
@dickeyxxx it is referenced here https://github.com/heroku/cli-engine/blob/726ed7f103813ff6dd0730b5f25d9ed11654baf1/src/plugins/cache.js#L159 and is the only way to force a save of the cache, I could refactor though.